client/my-sites/comments/comment/comment-link.jsx

```diff
@@ -0,0 +1,93 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import page from 'page';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import ExternalLink from 'components/external-link';
+import { isEnabled } from 'config';
+import { getSiteComment } from 'state/selectors';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+
+export class CommentLink extends PureComponent {
+	static propTypes = {
+		commentId: PropTypes.number,
+		isBulkMode: PropTypes.bool,
+	};
+
+	getFormattedDate = () => this.props.moment( this.props.commentDate ).format( 'll LT' );
+
+	getRelativeDate = () => {
+		const { commentDate, moment } = this.props;
+		return moment()
+			.subtract( 1, 'month' )
+			.isBefore( commentDate )
+			? moment( commentDate ).fromNow()
+			: moment( commentDate ).format( 'll' );
+	};
+
+	handleClick = event => {
+		if ( ! window ) {
+			return;
+		}
+		event.preventDefault();
+		window.scrollTo( 0, 0 );
+
+		const { commentId, siteSlug } = this.props;
+		const path = get( window, 'history.state.path' );
+
+		const newPath =
+			-1 !== path.indexOf( '#' )
+				? path.replace( /[#].*/, `#comment-${ commentId }` )
+				: `${ path }#comment-${ commentId }`;
+
+		window.history.replaceState( { ...window.history.state, path: newPath }, null );
+
+		page( `/comment/${ siteSlug }/${ commentId }` );
```

Comment:
I think that allowing components to have direct access to `page` is problematic. The navigation middleware has a `NAVIGATE` action that just calls `page( path )`. Maybe we can replace this with `dispatch( { type: NAVIGATE, path } )` if it doesn't break anything else.